gke-scripts/40-k8s-resources.sh

```diff
@@ -0,0 +1,77 @@
+#! /bin/bash
+
+function create_k8s_resources {
+  if [ $# -lt 4 ]
+  then
+    echo "usage: create_k8s_resources <service_name> <service_account_name> <port> <neg_name> <image_name> <optional-args-to-pass-to-binary>"
+    exit 1
+  fi
+
+  export SERVICE_NAME="$1"
+  export SERVICE_ACCOUNT_NAME="$2"
+  export SERVICE_PORT="$3"
+  export NEG_NAME="$4"
+  export SERVICE_IMAGE_NAME="$5"
+
+  shift 5
+
+  # Convert the optional args to be passed to the binary into a comma seperated
+  # string of arguments. Moving to this format makes it easier to specify as an
+  # array in the `args` field of the `container` spec in the deployment.
+  # TODO(easwars): See if there is a better way to do this.
+  i=1
+  export SERVICE_ARGS=
+  while [ $# -ne 0 ]
+  do
+    if [ $i -ne 1 ]
+    then
+      SERVICE_ARGS="${SERVICE_ARGS}, "
+    fi
+    SERVICE_ARGS="${SERVICE_ARGS}$1"
+    shift
+    i=$(( $i + 1 ))
+  done
+  echo $SERVICE_ARGS
+
+  envsubst < ServiceAccount.yaml | kubectl apply -f -
+
+  gcloud iam service-accounts add-iam-policy-binding \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[default/${SERVICE_ACCOUNT_NAME}]" \
+    ${PROJECT_NUM}-compute@developer.gserviceaccount.com
+
+  gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[default/${SERVICE_ACCOUNT_NAME}]" \
+    --role roles/trafficdirector.client
+
+  envsubst < Service.yaml | kubectl apply -f -
+  envsubst < Deployment.yaml | kubectl apply -f -
+
+  kubectl get svc ${SERVICE_NAME} -o yaml
+  kubectl get deployment ${SERVICE_NAME} -o yaml
+  sleep 10
```

Comment:
This might work: https://cloud.google.com/kubernetes-engine/docs/how-to/standalone-neg#validate_neg_creation
Something like `kubectl get svcneg` in a `while`.

Reply:
Sounds like a good idea but will need to be tested. If I have to retest for other changes I will also make this change.